sync/sync_head.go

```diff
@@ -99,9 +102,11 @@ func (s *Syncer[H]) subjectiveHead(ctx context.Context) (H, error) {
 	case isExpired(trustHead, s.Params.TrustingPeriod):
 		log.Warnw("subjective initialization with an expired header", "height", trustHead.Height())
 	case !isRecent(trustHead, s.Params.blockTime, s.Params.recencyThreshold):
+		s.metrics.laggingNetworkHead(s.ctx)
```

Comment:
This will be duplicated:
* first we mention it here.
* then in Head we do a recency check, and head calls the subjective head where the first happens

So we should remove this call